test/reproducers/checkAlgorithms/CheckAlgorithms.java

```diff
@@ -12,68 +8,82 @@ public class CheckAlgorithms {
     public static final String NONFIPS_ALGORITHM = "TLS_RSA_WITH_AES_128_CBC_SHA";
 
     public static void main(String[] args) throws Exception {
-	if (args.length != 2) {
-            System.err.println("Usage: CheckAlgorithms <true|false> <algorithms|providers>");
+        if (args.length != 2 || args[0].equals("--help") || args[0].equals("-h")) {
+            System.err.println("Test for listing available algorithms and providers and checking their FIPS compatibility");
+            System.err.println("Usage: CheckAlgorithms <true|false|only-algorithms|only-providers> <algorithms|providers|both>");
+            System.err.println("First argument: specify whether (and how) to check FIPS compatibility or not");
+            System.err.println("Second argument: specify what to check - algorithms, providers or both");
             System.exit(1);
         }
 
         // Parse the arguments
-        boolean shouldHonorFips = Boolean.parseBoolean(args[0]);
-        String testCategory = args[1];
+        String shouldHonorFips = args[0].toLowerCase();
+        String testCategory = args[1].toLowerCase();
 
-        // Check if the shouldHonorFips is "true" or "false"
-        if (!args[0].equalsIgnoreCase("true") && !args[0].equalsIgnoreCase("false")) {
+        // Check if the shouldHonorFips is valid value
+        List<String> possibleFirstArgs = Arrays.asList("true", "false", "only-algorithms", "only-providers");
+        if (!possibleFirstArgs.contains(shouldHonorFips)) {
             System.err.println("Invalid value for the first argument: " + args[0]);
             System.exit(1);
         }
 
-        // Check if the test category is "algorithms" or "providers"
-        if (!testCategory.equalsIgnoreCase("algorithms") && !testCategory.equalsIgnoreCase("providers")) {
-            System.err.println("Invalid test category: " + testCategory);
+        // Check if the test category is "algorithms", "providers" or "both"
+        List<String> possibleSecondArgs = Arrays.asList("algorithms", "providers", "both");
```

Comment:
This is duplicated with help. Please make it static final, immutable, and in the hel, stream.join it. it od not need to be perfect, but bbetter then duplicate it.